kafka-rest/src/main/java/io/confluent/kafkarest/resources/v3/TopicsResource.java

```diff
@@ -132,6 +134,37 @@ public void getTopic(
     AsyncResponses.asyncResume(asyncResponse, response);
   }
 
+  @PATCH
+  @Path("/{topicName}")
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Produces(MediaType.APPLICATION_JSON)
+  @PerformanceMetric("v3.topics.partitions")
+  @ResourceName("api.v3.topics.partitions")
+  public void updatePartitionCount(
+      @Suspended AsyncResponse asyncResponse,
+      @PathParam("clusterId") String clusterId,
+      @PathParam("topicName") String topicName,
+      @Valid PartitionCountRequest partitionCount) {
+
+    if (partitionCount == null) {
+      throw Errors.invalidPayloadException("Request body is empty. Partition_count is required.");
+    }
+
+    CompletableFuture<GetTopicResponse> response =
+        topicManager
+            .get()
+            .updateTopicPartitionCount(topicName, partitionCount.getPartitionCount())
+            .thenCompose(
+                nothing ->
+                    topicManager
+                        .get()
+                        .getTopic(clusterId, topicName)
+                        .thenApply(topic -> topic.orElseThrow(NotFoundException::new))
```

Comment:
Is this practically guaranteed not to throw if the update operation succeeded? It should still stay there to address edge cases like racing delete and update, I just want to understand the failure path.

Reply:
Yeah, I did go through that thought process, it's borrowed straight from the get topic code but I thought to leave it matching.  But you're right, there's only a really tiny timing window where the topic could be deleted.  Overkill though, so happy to take it off.